core/icons/mutator_icon.ts

```diff
@@ -0,0 +1,358 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as goog from '../../closure/goog/goog.js';
+goog.declareModuleId('Blockly.Mutator');
+
+import {Abstract} from '../events/events_abstract.js';
+import type {Block} from '../block.js';
+import {BlockChange} from '../events/events_block_change.js';
+import {BlocklyOptions} from '../blockly_options.js';
+import {BlockSvg} from '../block_svg.js';
+import type {Connection} from '../connection.js';
+import {Coordinate} from '../utils/coordinate.js';
+import * as dom from '../utils/dom.js';
+import * as eventUtils from '../events/utils.js';
+import {IHasBubble} from '../interfaces/i_has_bubble.js';
+import {Icon} from './icon.js';
+import {MiniWorkspaceBubble} from '../bubbles/mini_workspace_bubble.js';
+import {MUTATOR_TYPE} from './icon_types.js';
+import {Rect} from '../utils/rect.js';
+import {Size} from '../utils/size.js';
+import {Svg} from '../utils/svg.js';
+import type {WorkspaceSvg} from '../workspace_svg.js';
+
+export class MutatorIcon extends Icon implements IHasBubble {
+  /** The type string used to identify this icon. */
+  static readonly TYPE = MUTATOR_TYPE;
+
+  /**
+   * The weight this icon has relative to other icons. Icons with more positive
+   * weight values are rendered farther toward the end of the block.
+   */
+  static readonly WEIGHT = 1;
+
+  /** The size of this icon in workspace-scale units. */
+  private readonly SIZE = 17;
+
+  /** The distance between the root block and the workspace edges. */
+  private readonly MARGIN = 16;
```

Comment:
I marked them as private because I didn't see a reason for them to be accessed outside of the class, as opposed to `WEIGHT` which could be used in the `getWeight` of other icons. E.g.

```js
class MyCustomIcon implements IIcon {
  getWeight() {
    return MutatorIcon.WEIGHT + 1;
  }

  // Other IIcon methods.
}
```

`WEIGHT` is not intended to be modifyable, which is why I have marked it as `readonly`.

Does the style guide have opinions on how to specify constants (I couldn't find anything)? I don't like taking up space on every instance, but I also don't like cluttering up our exports with useless properties.